src/main/java/org/elasticsearch/search/aggregations/bucket/terms/support/IncludeExclude.java

```diff
@@ -325,11 +396,23 @@ private Automaton toAutomaton() {
     }
 
     public StringFilter convertToStringFilter() {
-        return new StringFilter(toAutomaton());
+        try {
+            return new AutomatonBackedStringFilter(toAutomaton());
+        } catch (TooComplexToDeterminizeException e) {
+            // Hard to pre-empt when this exception will occur due to the number
+            // terms and possible overlap of contents
+            // so we let it try and fall-back to Set based lookups if too
+            // complex.
+            return new TermListBackedStringFilter(includeValues, excludeValues);
+        }
```

Comment:
I'm wondering that maybe we should directly use TermListBackedStringFilter in that case: I wouldn't expect automatons to speed things up a lot and having TermListBackedStringFilter only in this catch block makes it hard to test.